crates/oxc_linter/src/rules/eslint/no_empty.rs

```diff
@@ -125,5 +188,29 @@ fn test() {
         ("try { foo(); } catch (ex) {} finally {}", None),
     ];
 
-    Tester::new(NoEmpty::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("try {} catch (ex) {throw ex}", "", None),
+        (
+            "try { foo() } catch (ex) {throw ex} finally {}",
+            "try { foo() } catch (ex) {throw ex} ",
+            None,
+        ),
+        // we can't fix this because removing the `catch` block would change the semantics of the code
+        ("try { foo() } catch (ex) {}", "try { foo() } catch (ex) {}", None),
+        ("if (foo) {}", "", None),
+        ("while (foo) {}", "", None),
+        ("for (;foo;) {}", "", None),
+        ("switch(foo) {}", "", None),
+        ("switch (foo) { /* empty */ }", "", None),
```

Comment:
Non-blocking: this should not be deleted

Reply:
hmm acaually i think this is correct (despite how strange it is)

https://github.com/eslint/eslint/blob/946ae00457265eb298eb169d6d48ca7ec71b9eef/tests/lib/rules/no-empty.js#L134-L142

eslint reports an error

Reply:
@DonIsaac i'm gonna merge this as is, if you think this is incorrect can you make an issue and assign this to me?

i think it makes sense to fix this in a separate PR as we currently report an error for this (maybe we shouldn't